__tests__/client/service-worker/events/utility.spec.js

```diff
@@ -0,0 +1,241 @@
+      // reset the property to match the metaBluePrint
+      meta[propName] = metaBluePrint[propName];
+    });
+  });
+});
+
+describe(createResourceMetaData.name, () => {
+  const appInfo = ['app', 'https://example.com/cdn/app/1.2.3-rc.4-abc123/'];
+  const appMetaData = {
+    type: 'one-app',
+    cacheName: 'one-app',
+    name: 'app',
+    bundle: 'browser',
+    version: '1.2.3-rc.4-abc123',
+  };
+  const moduleInfo = ['module', 'https://example.com/cdn/modules/test-root/2.2.2/'];
+  const baseMetaData = {
+    bundle: 'browser',
+    type: 'modules',
+    name: 'module',
+    version: '2.2.2',
+    cacheName: 'modules',
+    revision: '101010',
+  };
+  test.each([
+    // app
+    ['https://example.com/cdn/app/1.2.3-rc.4-abc123/app.js', appInfo, {
+      ...appMetaData,
+      path: 'app.js',
+    }],
+    ['https://example.com/cdn/app/1.2.3-rc.4-abc123/i18n/en-US.js', appInfo, {
+      ...appMetaData,
+      path: 'i18n/language.js',
+      locale: 'en-US',
+    }],
+    // modules
+    ['https://example.com/cdn/modules/test-root/2.2.2/test-root.browser.js', moduleInfo, {
+      ...baseMetaData,
+      path: 'test-root.browser.js',
+    }],
+    ['https://example.com/cdn/modules/test-root/2.2.2/test-root.legacy.browser.js', moduleInfo, {
+      ...baseMetaData,
+      bundle: 'legacy',
+      path: 'test-root.legacy.browser.js',
+    }],
+    ['https://example.com/cdn/modules/test-root/2.2.2/locale/en-US/test-root.json', moduleInfo, {
+      ...baseMetaData,
+      type: 'lang-packs',
+      path: 'en-US/test-root.json',
+      cacheName: 'lang-packs',
+      locale: 'en-US',
+    }],
+    ['https://example.com/cdn/modules/test-root/2.2.2/test-root.browser.js', moduleInfo, {
+      ...baseMetaData,
+      path: 'test-root.browser.js',
+      revision: null,
+    }],
+  ])('extracts metadata from %s', (url, [type, baseUrl], result) => {
+    // eslint-disable-next-line no-param-reassign
+    if (result.revision === null) delete result.revision;
+    const meta = createResourceMetaData({ request: { url } }, [type, baseUrl], result.revision);
+    expect(meta).toEqual({ ...result, url });
+  });
+});
+
+describe(invalidateCacheResource.name, () => {
+  beforeEach(() => {
+    jest.clearAllMocks();
+  });
+
+  test('gets the resource metadata and validates the incoming request', async () => {
+    const waitUntil = jest.fn();
+    const meta = { ...metaBluePrint };
+    const event = { request: { url: meta.url }, waitUntil };
+    const response = {};
+    const responseHandler = invalidateCacheResource(event, meta);
+
+    getMetaData.mockImplementationOnce(() => Promise.resolve({}));
+
+    expect(responseHandler(response)).toBe(response);
+    expect(waitUntil).toHaveBeenCalledTimes(1);
+    expect(getMetaData).toHaveBeenCalledTimes(1);
+    expect(getMetaData).toHaveBeenCalledWith({ cacheName: 'modules/assets/cat.jpg', url: 'http://localhost/modules/assets/cat.jpg' });
```

Comment:
`http://localhost/modules/assets/cat.jpg` does not appear to be a valid url. should this be the case ?